lib/puppet/provider/selinux_port/semanage.rb

```diff
@@ -5,7 +5,26 @@
   # SELinux must be enabled. Is there a way to get a better error message?
   confine selinux: true
 
-  python_command = Facter.value(:selinux_python_command)
+  # Find the correct version of python on the system
+  python_paths = [
+    '/usr/libexec/platform-python',
+    'python',
+    'python3',
+    'python2'
+  ]
+
+  python_command = nil
+  python_paths.each do |pypath|
+    candidate = Puppet::Util.which(pypath)
+
+    next unless candidate
+    if Puppet::Util::Execution.execute("#{candidate} -c 'import semanage'", failonfail: false).exitstatus.zero?
+      python_command = candidate
+      break
+    end
+  end
+  confine false: python_command.nil?
```

Comment:
I think this implementation means it also runs this code if the provider is not used. Is it possible to pass a proc to confine?

An alternative is to ship a wrapper (ba)sh script that calls exec at the end but I'm not sure if you can get the path right to a pluginsynced script.

Reply:
@ekohl Yeah, you can put a proc in a confine. I guess I can drop in an instance variable and see if I can set it in the proc.

Reply:
To be fair, the old fact ran every time as well so this isn't worse (it's just not better)

Reply:
I also had a look at `optional_command` but I'm not sure if that's better. Btw, how about this:

```ruby
def find_python_command
  python_paths.detect do |pypath|
    Puppet::Util::Execution.execute("#{pypath} -c 'import semanage'", failonfail: false).exitstatus.zero?
  end
end
```

I don't think you need to find an absolute path, the command should fail otherwise anyway, right?